enterprise/frontend/src/embedding-sdk/components/public/StaticDashboard/StaticDashboard.tsx

```diff
@@ -0,0 +1,85 @@
+import { pick } from "underscore";
+
+import {
+  DEFAULT_EMBED_DISPLAY_OPTIONS,
+  useDashboardFullscreen,
+  useDashboardRefreshPeriod,
+  useEmbedTheme,
+  useRefreshDashboard,
+} from "metabase/dashboard/hooks";
+import type {
+  EmbedDisplayParams,
+  RefreshPeriod,
+} from "metabase/dashboard/types";
+import { isNotNull } from "metabase/lib/types";
+import { PublicDashboard } from "metabase/public/containers/PublicDashboard/PublicDashboard";
+import { Box } from "metabase/ui";
+import type { DashboardId, ParameterId } from "metabase-types/api";
+
+export const StaticDashboard = ({
+  dashboardId,
+  parameterValues,
+  refreshPeriod: initialRefreshPeriod = null,
+  bordered,
+  titled,
+  theme: userTheme,
+  font,
+  hideDownloadButton,
+  hideParameters,
+}: {
+  dashboardId: DashboardId;
+  parameterValues: Record<ParameterId, string | string[] | null | undefined>;
+  refreshPeriod?: RefreshPeriod;
+} & Partial<EmbedDisplayParams>) => {
+  const options: EmbedDisplayParams = {
+    ...DEFAULT_EMBED_DISPLAY_OPTIONS,
+    ...pick(
+      {
+        bordered,
+        titled,
+        theme: userTheme,
+        font,
+        hideDownloadButton,
+        hideParameters,
+      },
+      isNotNull,
+    ),
+  };
+
+  const { refreshDashboard } = useRefreshDashboard({
+    dashboardId,
+    queryParams: parameterValues,
+  });
+  const { isFullscreen, onFullscreenChange, ref } = useDashboardFullscreen();
+  const { onRefreshPeriodChange, refreshPeriod, setRefreshElapsedHook } =
+    useDashboardRefreshPeriod({
+      onRefresh: refreshDashboard,
```

Comment:
API-wise, I just took a look at how `useDashboardRefreshPeriod` is used in another place. It seems `onRefresh` is also returned from `useRefreshDashboard`.

Would it make more sense to combine these 2 hooks
- `useRefreshDashboard` and
- `useDashboardRefreshPeriod`

if we don't really have to use `refreshDashboard` anywhere except passing it to `useDashboardRefreshPeriod`?

This could be a follow up task if we agree it should be refactored if changing it is too complicated here.

Reply:
Makes sense, let's do it in a separate PR? The idea was that I wouldn't be coupling the functionality together so I could debug things more easily and keep everything in a neat place. But if it's not going to be used anywhere else then :shrug:

Reply:
makes sense for another PR since that could add more noise to this PR.